common/containers/container.go

```diff
@@ -0,0 +1,224 @@
+		nextCont = nextCont[:i]
+		candidates = append(candidates, nextCont+"."+name)
+	}
+	candidates = append(candidates, name)
+	return c.candidatesWithAlias(candidates, name)
+}
+
+func (c *Container) candidatesWithAlias(candidates []string, name string) []string {
+	if len(c.aliases) == 0 {
+		return candidates
+	}
+	// If an alias exists for the name, ensure it is searched last.
+	alias, found := c.aliases[name]
+	if found {
+		return append(candidates, alias)
+	}
+	return candidates
+}
+
+// ContainerOption specifies a functional configuration option for a Container.
+type ContainerOption func(*Container) (*Container, error)
+
+// Aliases configures a set of simple names as aliases for fully-qualified names.
+//
+// An alias can be useful when working with variables, functions, and especially types from
+// multiple namespaces:
+//
+//    // CEL object construction
+//    qual.pkg.version.ObjTypeName{
+//       field: alt.container.ver.FieldTypeName{value: ...}
+//    }
+//
+// Only one qualified path may be used as CEL container, so at least one of these references is
+// a long qualified name within an otherwise short CEL program. Using the following aliases, the
+// program becomes much simpler:
+//
+//    // CEL Go option
+//    Aliases("qual.pkg.version.ObjTypeName", "alt.container.ver.FieldTypeName")
+//    // Simplified Object construction
+//    ObjTypeName{field: FieldTypeName{value: ...}}
+//
+// There are a few rules for the qualified names and the simple name aliases generated from them:
+// - Qualified names must be dot-delimited, e.g. `package.subpkg.name`.
+// - The last element in the qualified name is the simple name used as an alias.
+// - Alias names must not collide with each other.
+// - The alias name must not collide with the root-level container name.
+//
+// Aliases are distinct from container-based references in the following important ways:
+// - Containers follow C++ namespace resolution rules with searches from the most qualified name
+//   to the least qualified name.
+// - Container references within the CEL program may be relative, and are resolved to fully
+//   qualified names at either type-check time or program plan time, whichever comes first.
+// - Alias simple names must resolve to a fully-qualified name.
+// - Resolved aliases do not participate in namespace resolution.
+// - Resolved aliases are searched after container names, including container names in the global
+//   scope.
+//
+// If there is ever a case where an identifier could be in both the container and in the alias,
+// the container wins as the container will continue to evolve over time and the program must be
+// forward compatible with changes in the container.
+func Aliases(qualifiedNames ...string) ContainerOption {
+	return func(c *Container) (*Container, error) {
+		for _, qn := range qualifiedNames {
+			ind := strings.LastIndex(qn, ".")
+			alias := qn
+			if ind > 0 || ind < len(qn)-1 {
+				alias = qn[ind+1:]
+			}
```

Comment:
:bug: Did you mean to `&&` this test?
```suggestion
			if ind > 0 && ind < len(qn)-1 {
				alias = qn[ind+1:]
			}
```
As written, this test will almost always evaluate true.